micro-rdk/src/common/data_manager.rs

```diff
@@ -262,43 +252,128 @@ where
     async fn run<'b>(&mut self, app_client: &'b AppClient) -> Result<(), AppClientError> {
         for collector_key in self.resource_method_keys.iter() {
             let store_lock = self.store.lock().await;
-            let mut reader = match store_lock.get_reader(collector_key) {
-                Ok(reader) => reader,
-                Err(err) => {
-                    log::error!(
-                        "error acquiring reader for collector key ({:?}): {:?}",
-                        collector_key,
-                        err
-                    );
-                    continue;
-                }
-            };
-            let data = match read_messages_for_collector(&mut reader) {
-                Ok(data) => data,
-                Err(err) => {
+            loop {
+                let mut reader = match store_lock.get_reader(collector_key) {
+                    Ok(reader) => reader,
+                    Err(err) => {
+                        log::error!(
+                            "error acquiring reader for collector key ({:?}): {:?}",
+                            collector_key,
+                            err
+                        );
+                        continue;
```

Comment:
Does this mean to continue to the next collector? It looks to me like it will just immediately retry `get_reader(collector_key)`.  If you meant to continue with the next collector, just label the loops and then make clear on `break`s and `continue`s which one you are referring to.